keras_nlp/models/t5/t5_backbone.py

```diff
@@ -0,0 +1,251 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""T5 backbone model."""
+
+import copy
+
+import tensorflow as tf
+from tensorflow import keras
+
+from keras_nlp.api_export import keras_nlp_export
+from keras_nlp.layers.transformer_layer_utils import compute_causal_mask
+from keras_nlp.models.backbone import Backbone
+from keras_nlp.models.t5.t5_layer_norm import T5LayerNorm
+from keras_nlp.models.t5.t5_presets import backbone_presets
+from keras_nlp.models.t5.t5_transformer_layer import T5TransformerLayer
+from keras_nlp.utils.python_utils import classproperty
+
+
+@keras_nlp_export("keras_nlp.models.T5Backbone")
+class T5Backbone(Backbone):
+    """T5 encoder-decoder backbone model.
+
+    T5 is a LLM pretrained on a mix of unsupervised and supervised tasks,
+    where each task is converted to a sequence-to-sequence format.
+    T5 works well on a variety of tasks out-of-the-box by prepending
+    various prefixex to the input sequence, e.g., for translation:
+    `"translate English to German: ..."`, for summarization:
+    `"summarize: ..."`.
+
+    T5 was introduced in
+    [Exploring the Limits of Transfer Learning with a Unified Text-to-Text Transformer](https://arxiv.org/abs/1910.10683)
+
+    The default constructor gives a fully customizable, randomly initialized T5
+    model with any number of layers, heads, and embedding dimensions. To load
+    preset architectures and weights, use the `from_preset` constructor.
+
+    Disclaimer: Pre-trained models are provided on an "as is" basis, without
+    warranties or conditions of any kind. The underlying model is provided by
+    Google Research under the Apache 2.0 license
+    ([here](https://github.com/google-research/text-to-text-transfer-transformer/blob/main/LICENSE)).
+
+    Args:
+        vocabulary_size: int. The size of the token vocabulary.
+        num_layers: int. The number of Transformer layers.
+        num_heads: int. The number of attention heads for each Transformer.
+            The hidden size must be divisible by the number of attention heads.
+        hidden_dim: int. The hidden size of the Transformer layers.
+        intermediate_dim: int. The output dimension of the first Dense layer in
+            a two-layer feedforward network for each Transformer layer.
+        dropout: float. Dropout probability for the Transformer layers.
+        activation: activation function (or activation string name). The
+            activation to be used in the inner dense blocks of the
+            Transformer layers. Defaults to `"gelu"`. The original
+            T5 architecture used `"relu"`,
+            but the modern version uses `"gelu"`.
+        use_gated_activation: boolean. Whether to use activation gating in
+            the inner dense blocks of the Transformer layers.
+            Defaults to True. The original T5 architecture didn't use
+            gating, but the version does.
```

Comment:
but the version does -> but more recent versions do\

We could also link to https://arxiv.org/abs/2002.05202 , but maybe overkill

Reply:
Done.